src/app/App.tsx

```diff
@@ -136,20 +132,50 @@ function App(): JSX.Element {
   return (
     <BrowserRouter>
       <Routes>
+        <Route path="/" element={<Dashboard content={topics} />}></Route>
         <Route
-          path="/"
+          path="/:topicId"
           element={
-            <Dashboard
-              content={topics}
-              onTopicSubmit={handleTopicSubmit}
+            <TopicHandler
+              topics={topics}
               onNeedSubmit={handleNeedSubmit}
               onNeedUpvote={handleNeedUpvote}
             />
           }
         ></Route>
+        <Route
+          path="/addTopic"
+          element={<AddTopic onSubmit={handleTopicSubmit} />}
+        ></Route>
       </Routes>
     </BrowserRouter>
   )
 }
 
 export default App
+
+type TopicHandlerProps = {
+  topics: Topic[]
+  onNeedUpvote: (
+    topicId: string
+  ) => (needId: string) => (updatedVotes: number) => void
+  onNeedSubmit: (topicId: string) => (need: Need) => void
+}
+
+function TopicHandler({
+  topics,
+  onNeedUpvote,
+  onNeedSubmit,
+}: TopicHandlerProps): JSX.Element {
+  const params = useParams()
+  const topic = topics.find(topic => topic.id === params.topicId)
+  return topic ? (
+    <TopicView
+      onUpvoteChange={onNeedUpvote(topic.id)}
+      onNeedSubmit={onNeedSubmit(topic.id)}
+      content={topic}
+    />
+  ) : (
+    <h1>404 Page not found</h1>
```

Comment:
Same code as in PR#31?

Reply:
ups my bad, I guess its because I created this branch from the last page branch I used for the previous PR.